src/server/MapPlaylist.ts

```diff
@@ -63,17 +84,37 @@ export class MapPlaylist {
     }
   }
 
-  // Specifically controls how the playlists rotate.
+  private mapHistory: PlaylistType[] = [];
+
   private getNextPlaylistType(): PlaylistType {
-    switch (this.currentPlaylistCounter) {
-      case 0:
-      case 1:
-        this.currentPlaylistCounter++;
-        return PlaylistType.BigMaps;
-      case 2:
-        this.currentPlaylistCounter = 0;
-        return PlaylistType.SmallMaps;
+    // Keep approx. 2/3 BigMaps, 1/3 SmallMaps
+    const BIG_WEIGHT = 2;
+    const SMALL_WEIGHT = 1;
+    const MAX_HISTORY = 2;
+
+    const last = this.mapHistory[this.mapHistory.length - 1];
+    const secondLast = this.mapHistory[this.mapHistory.length - 2];
+
+    // Avoid repeating the same map type 3 times in a row
+    if (last === secondLast && last !== undefined) {
+      const opposite =
+        last === PlaylistType.BigMaps
+          ? PlaylistType.SmallMaps
+          : PlaylistType.BigMaps;
+      this.mapHistory.push(opposite);
+      if (this.mapHistory.length > MAX_HISTORY) this.mapHistory.shift();
+      return opposite;
     }
+
+    // Weighted random
+    const roll = Math.floor(Math.random() * (BIG_WEIGHT + SMALL_WEIGHT));
```

Comment:
floor has no effect on the outcome
```suggestion
    const roll = Math.random() * (BIG_WEIGHT + SMALL_WEIGHT);
```